include/GIL/Type.hpp

```diff
@@ -51,9 +53,24 @@ class Type {
     /// @return Returns a pointer to the type base.
     glu::types::TypeBase *getType() const { return _type; }
 
+    /// @brief Dereference operator.
+    /// @return A reference to the underlying TypeBase.
     glu::types::TypeBase &operator*() const { return *_type; }
 
+    /// @brief Arrow operator for accessing members of the underlying TypeBase.
+    /// @return A pointer to the underlying TypeBase.
     glu::types::TypeBase *operator->() const { return _type; }
+
+    /// @brief Checks if two types are equal.
+    /// @param other The other type to compare with.
+    /// @return True if both types have the same size, alignment, constness, and
+    /// base type.
+    bool operator==(Type const &other) const
+    {
+        return _fields.size == other._fields.size
+            && _fields.alignment == other._fields.alignment
+            && _fields.isConst == other._fields.isConst && _type == other._type;
```

Comment:
this should really be
```suggestion
        return _type == other._type;
```
all other fields are calculated from the type, so if the pointer is the same, the metadata has to be the same